core/src/main/scala/com/mesosphere/usi/core/SchedulerLogicHandler.scala

```diff
@@ -118,15 +118,38 @@ private[core] class SchedulerLogicHandler(mesosCallFactory: MesosCalls) {
     frameResultBuilder.result
   }
 
+  def generateSuppressCalls(pendingLaunch: Set[PodId], newLaunched: Set[PodId]): List[Call] = {
+    val alreadyLaunchedRoles = newLaunched.iterator.collect {
+      case id if specs.podSpecs.contains(id) => specs.podSpecs(id).runSpec.role
+    }
+
+    val rolesBeingLaunched = pendingLaunch
+      .map(id => specs.podSpecs.get(id))
+      .collect { case Some(p) => p.runSpec.role }
+
+    alreadyLaunchedRoles
+      .filterNot(rolesBeingLaunched)
```

Comment:
Could you please add types to  `rolesBeingLaunched` and `alreadyLaunchedRoles`. Otherwise `.filterNot(rolesBeingLaunched)` looks a bit like code-golf ;)